src/MANYBODY/pair_eam.cpp

```diff
@@ -933,5 +933,19 @@ void *PairEAM::extract(const char *str, int &dim)
 {
   dim = 2;
   if (strcmp(str,"scale") == 0) return (void *) scale;
+
+  // peratom requests from FixPair
```

Comment:
I think it is a very bad idea to have different meanings for "dim" depending on which function may call extract(). The extract() function has no way to figure it out who is calling and if a property is available, it may be used elsewhere.  Either there would need to be a different function, say extract_peratom() for that purpose that can have "cols" instead of "dim" as argument or there should be an additional property, e.g. rho_dims providing the missing size information in each of the "dim" dimensions.

Reply:
yes - added an extract_peratom() method